File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.ts
function largestAltitude(gain: number[]): number {

  let curr_alt: number = 0;

  let max_alt: number = gain.reduce((max_alt, alt) => {
    curr_alt += alt;
    return Math.max(curr_alt, max_alt);
  }, 0);

  return max_alt;  
};

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        
        int currAltitude = 0;
        int maxAltitude = 0;

        for(int elem : gain) {

            currAltitude = elem + currAltitude;
            maxAltitude = Math.max(maxAltitude, currAltitude);
        }

        return maxAltitude;
    }
}